src/coreclr/jit/lsra.cpp

```diff
@@ -887,6 +887,17 @@ void LinearScan::setBlockSequence()
             }
         }
 
+#if defined(FEATURE_EH_FUNCLETS) && defined(TARGET_ARM)
+        // For ARM, if this block is BBCallAlwaysPairTail, then it won't have any preds
+        // since we use "lr" mechanism to jump to the block that is executed after try block is done.
+        // In such case, mark that we do not want to insert resolution moves in it.
+        if (block->bbPreds == nullptr && block->isBBCallAlwaysPairTail())
+        {
+            blockInfo[block->bbNum].hasEHBoundaryIn  = true;
```

Comment:
You might note this is a bit of a subterfuge, as this block has no flow whatsoever.

Reply:
Yes, agree and since we never had a flow, we were not entering the for loop above. However, thought it would be good to have the logic (that is inside for loop for `isBBCallAlwaysPairTail()`) outside the loop for no flow scenarios.